modules/msg/src/cfe_msg_ccsdspri.c

```diff
@@ -0,0 +1,352 @@
+
+    return CFE_SUCCESS;
+}
+
+/******************************************************************************
+ * Get message type - See API and header file for details
+ */
+int32 CFE_MSG_GetType(const CFE_MSG_Message_t *MsgPtr, CFE_MSG_Type_t *Type)
+{
+
+    if (MsgPtr == NULL || Type == NULL)
+    {
+        return CFE_MSG_BAD_ARGUMENT;
+    }
+
+    if ((MsgPtr->CCSDS.Pri.StreamId[0] & (CFE_MSG_TYPE_MASK >> 8)) != 0)
+    {
+        *Type = CFE_MSG_Type_Cmd;
+    }
+    else
+    {
+        *Type = CFE_MSG_Type_Tlm;
+    }
+
+    return CFE_SUCCESS;
+}
+
+/******************************************************************************
+ * Set message type - See API and header file for details
+ */
+int32 CFE_MSG_SetType(CFE_MSG_Message_t *MsgPtr, CFE_MSG_Type_t Type)
+{
+    int32 status = CFE_SUCCESS;
+
+    if (MsgPtr == NULL)
+    {
+        return CFE_MSG_BAD_ARGUMENT;
+    }
+
+    if (Type == CFE_MSG_Type_Cmd)
+    {
+        MsgPtr->CCSDS.Pri.StreamId[0] |= CFE_MSG_TYPE_MASK >> 8;
+    }
+    else if (Type == CFE_MSG_Type_Tlm)
+    {
+        MsgPtr->CCSDS.Pri.StreamId[0] &= ~(CFE_MSG_TYPE_MASK >> 8);
+    }
+    else
+    {
+        status = CFE_MSG_BAD_ARGUMENT;
+    }
+
+    return status;
+}
+
+/******************************************************************************
+ * Get message has secondary header flag - See API and header file for details
+ */
+int32 CFE_MSG_GetHasSecondaryHeader(const CFE_MSG_Message_t *MsgPtr, bool *HasSecondary)
+{
+
+    if (MsgPtr == NULL || HasSecondary == NULL)
+    {
+        return CFE_MSG_BAD_ARGUMENT;
+    }
+
+    *HasSecondary = (MsgPtr->CCSDS.Pri.StreamId[0] & (CFE_MSG_SHDR_MASK >> 8)) != 0;
+
+    return CFE_SUCCESS;
+}
+
+/******************************************************************************
+ * Set message has secondary header flag - See API and header file for details
+ */
+int32 CFE_MSG_SetHasSecondaryHeader(CFE_MSG_Message_t *MsgPtr, bool HasSecondary)
+{
+    if (MsgPtr == NULL)
+    {
+        return CFE_MSG_BAD_ARGUMENT;
+    }
+
+    if (HasSecondary)
+    {
+        MsgPtr->CCSDS.Pri.StreamId[0] |= CFE_MSG_SHDR_MASK >> 8;
+    }
+    else
+    {
+        MsgPtr->CCSDS.Pri.StreamId[0] &= ~(CFE_MSG_SHDR_MASK >> 8);
+    }
+
+    return CFE_SUCCESS;
+}
+
+/******************************************************************************
+ * Get message application ID - See API and header file for details
+ */
+int32 CFE_MSG_GetApId(const CFE_MSG_Message_t *MsgPtr, CFE_MSG_ApId_t *ApId)
+{
+
+    if (MsgPtr == NULL || ApId == NULL)
+    {
+        return CFE_MSG_BAD_ARGUMENT;
+    }
+
+    CFE_MSG_GetHeaderField(MsgPtr->CCSDS.Pri.StreamId, ApId, CFE_MSG_APID_MASK);
+
+    return CFE_SUCCESS;
+}
+
+/******************************************************************************
+ * Set message application ID - See API and header file for details
+ */
+int32 CFE_MSG_SetApId(CFE_MSG_Message_t *MsgPtr, CFE_MSG_ApId_t ApId)
```

Comment:
For "setter" API's, it might be worth considering passing by pointer (e.g. `const CFE_MSG_ApId_t*` rather than by value.  A pattern of passing by value kind of assumes that each logical "field" is just a single datum, like an int or a float.  If this becomes a struct or string this could get very inefficient.  But if it were a pointer then it could accept any data type.